k8sdeps/kv/plugin/builtin/literals.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package builtin
+
+import (
+	"fmt"
+	"strings"
+
+	"sigs.k8s.io/kustomize/k8sdeps/kv"
+)
+
+// Literals takes a list of literals.
+// Each literal source should be a key and literal value,
+// e.g. `somekey=somevalue`
+// It will be similar to kubectl create configmap|secret --from-literal
+type Literals struct{}
+
+// Get implements the interface for kv plugins.
+func (p Literals) Get(root string, args []string) ([]kv.Pair, error) {
+	var kvs []kv.Pair
+	for _, s := range args {
+		k, v, err := parseLiteralSource(s)
+		if err != nil {
+			return nil, err
+		}
+		kvs = append(kvs, kv.Pair{Key: k, Value: v})
+	}
+	return kvs, nil
+}
+
+// parseLiteralSource parses the source key=val pair into its component pieces.
+// This functionality is distinguished from strings.SplitN(source, "=", 2) since
+// it returns an error in the case of empty keys, values, or a missing equals sign.
+func parseLiteralSource(source string) (keyName, value string, err error) {
```

Comment:
Until we decide to remove the non-plugin version of literals KV generation in k8sdeps/configmapandsecret/basefactory.go, we should use the same methods there and here.  This will also be true for KV gen from env and files.

Could make the methods in configmapandsecret public, and use them here from there, or move them here and do the opposite, or slap them into a utils package...

Reply:
Ok, ill update.